fop/fom.c

```diff
@@ -634,6 +638,12 @@ M0_INTERNAL void m0_fom_queue(struct m0_fom *fom)
 	fom->fo_loc = dom->fd_localities[loc_idx];
 	fom->fo_loc_idx = loc_idx;
 	m0_fom_sm_init(fom);
+	m0_atomic64_inc(&fom->fo_service->rs_fom_queued);
+	if (m0_reqh_service_state_get(fom->fo_service) == M0_RST_STOPPED) {
+		m0_fom_phase_set(fom, M0_FOM_PHASE_FINISH);
+		m0_fom_fini(fom);
+		return;
+	}
```

Comment:
BUG? You increase the counter and then return without decreasing it? Shouldn't it be increased after the `if` block?

Reply:
Here service is stopped already, which contains this variable. As there is no lock held, service stop check is added.

Reply:
I agree with @andriytk . I think this is a bug.

Reply:
Okay, now counter is incremented after service check is done.

Reply:
This is fine with me.